internal/storage/storage.go

```diff
@@ -113,5 +140,85 @@ func (m *FileStorage) Get(short string) string {
 		fmt.Println("Ошибка сканирования файла:", err)
 	}
 
-	return result[short].OriginalURL
+	return result[short].OriginalURL, nil
+}
+
+func (m *FileStorage) GetShort(key string) (string, error) {
+	return "", nil
+}
+
+type DatabaseStorage struct {
+	db *sql.DB
+	mu sync.RWMutex
+}
+
+func NewDatabaseStorage(db *sql.DB) *DatabaseStorage {
+	return &DatabaseStorage{
+		db: db,
+	}
+}
+
+func (m *DatabaseStorage) Add(url string, short string) (err error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	id := uuid.New()
+	_, err = m.db.ExecContext(context.Background(), "INSERT INTO link (uuid, original_url, short_url) VALUES ($1, $2, $3)", id, url, short)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *DatabaseStorage) AddBatch(urls []string, shorts []string) (err error) {
```

Comment:
Не критично, но в случае одного возвращаемого значения нет смысла использовать именованый.
```suggestion
func (m *DatabaseStorage) AddBatch(urls []string, shorts []string) error {
```